src/core/main.c

```diff
@@ -2603,9 +2580,13 @@ int main(int argc, char *argv[]) {
         (void) reset_all_signal_handlers();
         (void) ignore_signals(SIGNALS_IGNORE, -1);
 
-        r = load_configuration(argc, argv, &saved_rlimit_nofile, &saved_rlimit_memlock, &error_message);
-        if (r < 0)
+        (void) parse_configuration(&saved_rlimit_nofile, &saved_rlimit_memlock);
+
+        r = parse_argv(argc, argv);
+        if (r < 0) {
+                error_message = "Failed to parse commandline arguments";
                 goto finish;
+        }
```

Comment:
ahum, you are basically undoing 97d1fb94ba3b557abf9acbcd47acb11dbeb1239d, i.e. the excercise of making main() a bit less massive...

Reply:
As described in the commit message, load_configuration() slightly obfuscates the code but if you prefer I can just drop this patch, I don't really mind.

Reply:
97d1fb9 was moving quite a lot of stuff out, and this patch is only adding back a few lines. I think it is worth it, esp. that it gets rid of the need to pass `error_message`.